src/main/java/taxi/filter/AuthenticationFilter.java

```diff
@@ -0,0 +1,42 @@
+package taxi.filter;
+
+import java.io.IOException;
+import java.util.HashSet;
+import java.util.Set;
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.annotation.WebFilter;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebFilter(urlPatterns = "/*")
+public class AuthenticationFilter implements Filter {
+    private final Set<String> allowedUrls = new HashSet<>();
+
+    @Override
+    public void init(FilterConfig filterConfig) {
+        allowedUrls.add("/login");
+        allowedUrls.add("/drivers/add");
+    }
+
+    @Override
+    public void doFilter(ServletRequest request, ServletResponse response, FilterChain filterChain)
+            throws IOException, ServletException {
+        HttpServletRequest req = (HttpServletRequest) request;
+        HttpServletResponse res = (HttpServletResponse) response;
+        if (allowedUrls.contains(req.getServletPath())) {
+            filterChain.doFilter(req, res);
+            return;
+        }
+        Long driverId = (Long) req.getSession().getAttribute("driver_id");
+        if (driverId == null) {
+            res.sendRedirect(req.getContextPath() + "/login");
+            return;
+        }
```

Comment:
<img width="737" alt="image" src="https://user-images.githubusercontent.com/58032907/204477415-2efe6626-c979-426f-a94b-f2d5472826d5.png">

Reply:
@sequencerr think about this